clustering_bounds_comparator.hh

```diff
@@ -41,6 +41,8 @@ enum class bound_kind : uint8_t {
 std::ostream& operator<<(std::ostream& out, const bound_kind k);
 
 bound_kind invert_kind(bound_kind k);
+// Swaps start <-> end
+bound_kind reverse_kind(bound_kind k);
```

Comment:
Now that you have both invert_kind() and reverse_kind(), please also add a comment to invert_kind() to make it clearer what's the difference.